irksome/getForm.py

```diff
@@ -124,60 +142,81 @@ def getForm(F, butch, t, dt, u0, bcs=None, bc_type="DAE"):
          either `f.interpolate(expr-c*u0)` or `f.project(expr-c*u0)`, depending
          on whether the function space for f supports interpolation or
          not.
-
     """
     v = F.arguments()[0]
     V = v.function_space()
     assert V == u0.function_space()
 
-    A = numpy.array([[Constant(aa) for aa in arow] for arow in butch.A],
-                    dtype=object)
     c = numpy.array([Constant(ci) for ci in butch.c],
                     dtype=object)
 
-    num_stages = len(c)
+    bA1, bA2 = splitting(butch.A)
+
+    try:
+        bA1inv = numpy.linalg.inv(bA1)
+    except numpy.linalg.LinAlgError:
+        bA1inv = None
+    try:
+        bA2inv = numpy.linalg.inv(bA2)
+    except numpy.linalg.LinAlgError:
+        bA2inv = None
+
+    A1 = numpy.array([[Constant(aa) for aa in arow] for arow in bA1],
+                     dtype=object)
+    if bA1inv is not None:
+        A1inv = numpy.array([[ConstantOrZero(aa) for aa in arow] for arow in bA1inv],
+                            dtype=object)
+    else:
+        A1inv = None
+    A2inv = numpy.array([[ConstantOrZero(aa) for aa in arow] for arow in bA2inv],
+                        dtype=object)
```

Comment:
`bA2inv` may be `None` at this point, so you could get a `TypeError` here.

Reply:
Since you always need `A2inv` probably you want to say
```
if bA2inv is None:
    raise NotImplementedError("... is not invertible, can't continue")
```

Reply:
I think that we require `A2inv` everywhere.  So you're right, we should just error when we try inverting `bA2`.